sw/host/opentitanlib/src/image/manifest_def.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::image::manifest::*;
+use crate::util::bigint::fixed_size_bigint;
+use crate::util::num_de::HexEncoded;
+use crate::util::parse_int::ParseInt;
+
+use anyhow::{anyhow, Result};
+use serde::Deserialize;
+use std::convert::{TryFrom, TryInto};
+use std::iter::IntoIterator;
+
+fixed_size_bigint!(ManifestRsa, 3072);
+
+#[derive(Debug, Deserialize)]
+struct ManifestBigInt(Option<HexEncoded<ManifestRsa>>);
+
+#[derive(Debug, Deserialize)]
+struct ManifestSmallInt<T: ParseInt>(Option<HexEncoded<T>>);
+
+/// A macro for wrapping manifest struct definitions that parse from HJSON.
+///
+/// The #[repr(C)] version of `Manifest` can only be built when the fields in `ManifestDef` are
+/// present. This macro sets up the field by field conversion and provides the field names for
+/// purposes of error reporting.
+macro_rules! manifest_def {
+    (struct $name:ident {
+        $($field_name:ident: $field_type:ty,)*
+    }, $out_type:ident) => {
+        #[derive(Deserialize, Debug)]
+        struct $name {
+            $($field_name: $field_type,)*
+        }
+
+        impl ManifestPacked<$out_type> for $name {
+            fn unpack(self, _name: &'static str) -> Result<$out_type> {
+                Ok($out_type {
+                    // Call `unpack()` on each field with the field's name included for use in
```

Comment:
Would you mind wrapping this to 80 columns or whatever by hand? rustfmt doesn;t like formatting the insides of macros.